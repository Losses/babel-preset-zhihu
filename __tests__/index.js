import {transform} from '@babel/core'

describe('transform', () => {
  const defaultOptions = {
    envName: 'development',
    configFile: false,
    presets: ['module:.'],
  }

  const t = (code, options = {}) =>
    transform(code, {...defaultOptions, ...options})
      .code.split(process.cwd())
      .join('<cwd>')

  test('normal', () => {
    expect(t(`const fn = () => {}`)).toMatchSnapshot()
  })

  test('inject runtime', () => {
    expect(t(`async function* fn() {}`)).toMatchSnapshot()
  })

  test('compile modules', () => {
    expect(t(`import React from 'react'`)).toMatchSnapshot()
    expect(
      t(`import React from 'react'`, {
        caller: {name: 'test', supportsStaticESM: true},
      }),
    ).toMatchSnapshot()
  })
})
