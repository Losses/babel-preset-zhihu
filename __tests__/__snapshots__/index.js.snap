// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transform compile modules 1`] = `
"\\"use strict\\";

var _interopRequireDefault = require(\\"<cwd>/node_modules/@babel/runtime/helpers/interopRequireDefault\\");

var _react = _interopRequireDefault(require(\\"react\\"));"
`;

exports[`transform compile modules 2`] = `"import React from 'react';"`;

exports[`transform inject runtime 1`] = `
"\\"use strict\\";

var _interopRequireDefault = require(\\"<cwd>/node_modules/@babel/runtime/helpers/interopRequireDefault\\");

var _regenerator = _interopRequireDefault(require(\\"<cwd>/node_modules/@babel/runtime/regenerator\\"));

var _awaitAsyncGenerator2 = _interopRequireDefault(require(\\"<cwd>/node_modules/@babel/runtime/helpers/awaitAsyncGenerator\\"));

var _wrapAsyncGenerator2 = _interopRequireDefault(require(\\"<cwd>/node_modules/@babel/runtime/helpers/wrapAsyncGenerator\\"));

function fn() {
  return _fn.apply(this, arguments);
}

function _fn() {
  _fn = (0, _wrapAsyncGenerator2.default)(
  /*#__PURE__*/
  _regenerator.default.mark(function _callee() {
    return _regenerator.default.wrap(function _callee$(_context) {
      while (1) {
        switch (_context.prev = _context.next) {
          case 0:
          case \\"end\\":
            return _context.stop();
        }
      }
    }, _callee, this);
  }));
  return _fn.apply(this, arguments);
}"
`;

exports[`transform normal 1`] = `
"\\"use strict\\";

var fn = function fn() {};"
`;
